java/lang/StringBuilder/StringBuilder.cpp

```diff
@@ -0,0 +1,868 @@
+}
+
+StringBuilder StringBuilder::append(const CharSequence &target) {
+    string targetString = target.toString();
+    return this->append(targetString);
+}
+
+StringBuilder StringBuilder::append(const CharSequence &target, int start, int end) {
+    string targetString = target.toString();
+    int lengthOfTarget = (int)strlen(targetString);
+    if (start < 0 || start > end || end > lengthOfTarget) {
+        throw IndexOutOfBoundsException();
+    }
+
+    int numberOfCharacter = end - start;
+    int newLength = currentLength + numberOfCharacter;
+    this->ensureCapacity(newLength);
+    int indexOfTarget;
+    int indexOfOrginal;
+    for (indexOfTarget = start, indexOfOrginal = this->currentLength; indexOfTarget < end; indexOfTarget++, indexOfOrginal++) {
+        this->original[indexOfOrginal] = targetString[indexOfTarget];
+    }
+
+    this->currentLength = newLength;
+    return *this;
+}
+
+StringBuilder StringBuilder::append(const std::initializer_list<char> &target) {
+    int newLength = this->currentLength + (int)target.size();
+    this->ensureCapacity(newLength);
+    int index = this->currentLength;
+    std::initializer_list<char>::const_iterator listIterator;
+    for (listIterator = target.begin(); listIterator != target.end(); listIterator++) {
+        this->original[index] = *listIterator;
+        index = index + 1;
+    }
+    this->currentLength = newLength;
+    return *this;
+}
+
+StringBuilder StringBuilder::append(const string target) {
+    int stringLength = (int)strlen(target);
+    int newLength = this->currentLength + stringLength;
+    this->ensureCapacity(newLength);
+    int indexOfOriginal;
+    int indexOfString = 0;
+    for (indexOfOriginal = this->currentLength; indexOfOriginal < newLength; indexOfOriginal++) {
+        this->original[indexOfOriginal] = target[indexOfString];
+        indexOfString = indexOfString + 1;
+    }
+    this->currentLength = newLength;
+    return *this;
+}
+
+StringBuilder StringBuilder::append(const Double &target) {
+    return this->append(target.toString());
+}
+
+StringBuilder StringBuilder::append(double target) {
+    return this->append(Double(target));
+}
+
+StringBuilder StringBuilder::append(const Float &target) {
+    return this->append(target.toString());
+}
+
+StringBuilder StringBuilder::append(float target) {
+    return this->append(Float(target));
+}
+
+StringBuilder StringBuilder::append(const Integer &target) {
+    return this->append(target.toString());
+}
+
+StringBuilder StringBuilder::append(int target) {
+    return this->append(Integer(target));
+}
+
+StringBuilder StringBuilder::append(const Long &target) {
+    return this->append(target.toString());
+}
+
+StringBuilder StringBuilder::append(long target) {
+    return this->append(Long(target));
+}
+
+StringBuilder StringBuilder::append(const String &target) {
+    return this->append(target.toString());
+}
+
+StringBuilder StringBuilder::appendCodePoint(int codePoint) {
+    // TODO
```

Comment:
This method need some methods from Character class, but they have not been implemented yet.